### PR TITLE
Sonar refactor 2nd try

### PR DESCRIPTION
**Pull request by brobi1999**
sonar refactors for TorpedoStore.java

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -12,6 +12,8 @@
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
 
+  Random generator = new Random();
+
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Should be private

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: OK

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
...
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:41]: OK

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+    
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:46]: OK

**Review by R4zor25 — CHANGES_REQUESTED** [review]
Waiting for refactor

**Review by R4zor25 — APPROVED** [review]
Approved

**Comment by brobi1999** [comment]
Refactors successful

**Pull request merged**